### ISSUE TITLE
Semantic Segmentation

### ISSUE THREAD

**Issue by TheBigTicket02** [catalyst-team/catalyst#895]
## ❓ Questions and Help
Hi. I've started using Catalst only this week, so question could be stupid.
#### What have you tried?
I’ve tried to change *segmentation-tutorial.ipynb* for my project, where I have 20 classes.
I’ve changed .gif -> .png. and gif_imread -> utils.imread and albumentations.
show_random works as expected. 
model = smp.FPN(encoder_name="resnext50_32x4d", classes=1) 1->20
BCEWithLogitsLoss -> CrossEntropyLoss()
and when I ran runner
#### Code
   
`RuntimeError The size of tensor a (3) must match the size of tensor b (864) at non-singleton dimension 4`

runner.input[’image’].shape: torch.Size([2, 3, 864, 864])
runner.input[’mask’].shape: torch.Size([2, 1, 864, 864, 3])
runner.output[’logits’].shape: torch.Size([2, 20, 864, 864])

#### What is your question?
How can I change 'mask'.shape?

#### What's your environment?
Kaggle's latest environment

**Comment by le-Greg** [comment]
Hello,
I am currently doing more or less the same stuff, i think your mask should contain a 864*864 matrix of numbers from 0 to 19, to represent the different classes. Here you have a weird 5th dimension of size 3 at the end, maybe it is because your mask is still containing values from 0 to 255 in RGB ?

**Comment by Scitator** [comment]
Hi, **very** minimal _working_ example
```python
import os
import torch
from torch import nn
from torch.nn import functional as F
from torch.utils.data import DataLoader
from catalyst import dl
from catalyst.contrib.data.transforms import ToTensor
from catalyst.contrib.datasets import MNIST
from catalyst.utils import metrics

class NotUnet(nn.Module):

    def __init__(self, in_channels, num_segmentaiton_classes):
        super().__init__()
        self.encoder = nn.Sequential(nn.Conv2d(in_channels, in_channels, 3, 1, 1), nn.Tanh())
        self.decoder = nn.Conv2d(in_channels, num_segmentaiton_classes, 3, 1, 1)

    def forward(self, x):
        z = self.encoder(x)
        x_ = self.decoder(z)
        return x_

model = NotUnet(1, 2)
optimizer = torch.optim.Adam(model.parameters(), lr=0.02)

loaders = {
    "train": DataLoader(MNIST(os.getcwd(), train=False, download=True, transform=ToTensor()), batch_size=32),
    "valid": DataLoader(MNIST(os.getcwd(), train=False, download=True, transform=ToTensor()), batch_size=32),
}

class CustomRunner(dl.Runner):

    def _handle_batch(self, batch):
        x, _ = batch
        x_noise = (x + torch.rand_like(x)).clamp_(0, 1)
        x_ = self.model(x_noise)
        loss = F.cross_entropy(x_, x.to(torch.int64).squeeze(1))
        self.batch_metrics = {"loss": loss}
        
        if self.is_train_loader:
            loss.backward()
            self.optimizer.step()
            self.optimizer.zero_grad()

runner = CustomRunner()
runner.train(
    model=model, 
    optimizer=optimizer, 
    loaders=loaders, 
    verbose=True,
)
```

https://colab.research.google.com/drive/1Z9Ka5KLs_7P8XQYNIwL1c0xJzOWJbeHd?usp=sharing

**Comment by TheBigTicket02** [comment]
Thanks. It's working now.
